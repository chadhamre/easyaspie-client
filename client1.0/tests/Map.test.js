import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Map from '../components/MapComponent'

const map = renderer.create(<Map />)
const mapInstance = map.root;
const rendered = map.toJSON();
const mapShallow = shallow(
    <Map />
  );
const mapRender = mapShallow.dive();

it('MapComponent renders without crashing', () => {
  expect(rendered).toBeTruthy();
});

it('MapComponent renders correctly', () => {
  expect(rendered).toMatchSnapshot();
});

it('MapComponent should not be affected by props', () => {
  const mapShallow = shallow(
    <Map foo='bar'/>
  );
  expect(mapShallow).toMatchSnapshot();
  mapShallow.setProps({random:'prop'});
  expect(mapShallow).toMatchSnapshot();
});

describe('Testing render of MapView', () => {
  it('should not render a MapView if no initial location', () => {
    expect(mapRender.find('MapView').length).toBe(0)
  })
  it('should render a MapView if there is a location', () => {
    mapShallow.setState({location: {coords: {longitude: 1, latitude: 1}}})
    expect(mapShallow.dive().find('MapView').length).toBe(1);
  })
});

it('MapComponent should always have a RestaurantModal', () => {
  expect(mapRender.find('RestaurantModal').length).toBe(1);
  mapShallow.setState({location: {coords: {longitude: 1, latitude: 1}}})
  expect(mapShallow.dive().find('RestaurantModal').length).toBe(1);
});

