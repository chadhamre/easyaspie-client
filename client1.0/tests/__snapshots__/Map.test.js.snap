// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
     Loading 
  </Text>
  <View />
</View>
`;

exports[`MapComponent should not be affected by props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    foo="bar"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
>
           Loading 
</Text>,
        <RestaurantModal
          triggerModal={undefined}
/>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " Loading ",
          "disabled": false,
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": " Loading ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "triggerModal": undefined,
        },
        "ref": "modal",
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
>
             Loading 
</Text>,
          <RestaurantModal
            triggerModal={undefined}
/>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " Loading ",
            "disabled": false,
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": " Loading ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "triggerModal": undefined,
          },
          "ref": "modal",
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MapComponent should not be affected by props 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    foo="bar"
    random="prop"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
>
           Loading 
</Text>,
        <RestaurantModal
          triggerModal={undefined}
/>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " Loading ",
          "disabled": false,
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": " Loading ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "triggerModal": undefined,
        },
        "ref": "modal",
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
>
             Loading 
</Text>,
          <RestaurantModal
            triggerModal={undefined}
/>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " Loading ",
            "disabled": false,
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": " Loading ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "triggerModal": undefined,
          },
          "ref": "modal",
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
